encoding/csv.ts

```diff
@@ -361,44 +353,22 @@ export interface ParseOptions extends ReadOptions {
    * If you provide `string[]` or `ColumnOptions[]`, those names will be used for header definition.
    */
   columns?: string[] | ColumnOptions[];
-
-  /** Parse function for rows.
-   * Example:
-   * ```ts
-   *     import { parse } from "./csv.ts";
-   *     const r = await parse('a,b,c\ne,f,g\n', {
-   *      columns: ["this", "is", "sparta"],
-   *       parse: (_e: unknown) => {
-   *         const e = _e as { this: unknown, is: unknown, sparta: unknown };
-   *         return { super: e.this, street: e.is, fighter: e.sparta };
-   *       }
-   *     });
-   * // output
-   * // [
-   * //   { super: "a", street: "b", fighter: "c" },
-   * //   { super: "e", street: "f", fighter: "g" }
-   * // ]
-   * ```
-   */
-  parse?: (input: unknown) => unknown;
 }
 
 /**
  * Csv parse helper to manipulate data.
- * Provides an auto/custom mapper for columns and parse function
- * for columns and rows.
+ * Provides an auto/custom mapper for columns.
  * @param input Input to parse. Can be a string or BufReader.
  * @param opt options of the parser.
- * @returns If you don't provide `opt.skipFirstRow`, `opt.parse`, and `opt.columns`, it returns `string[][]`.
- *   If you provide `opt.skipFirstRow` or `opt.columns` but not `opt.parse`, it returns `object[]`.
- *   If you provide `opt.parse`, it returns an array where each element is the value returned from `opt.parse`.
+ * @returns If you don't provide `opt.skipFirstRow` and `opt.columns`, it returns `string[][]`.
+ *   If you provide `opt.skipFirstRow` or `opt.columns`, it returns `Record<string, unkown>[]`.
  */
 export async function parse(
   input: string | BufReader,
   opt: ParseOptions = {
     skipFirstRow: false,
   },
-): Promise<unknown[]> {
+): Promise<string[][] | Record<string, unknown>[]> {
```

Comment:
I think if we do overloading like the below, it would be able to return more correct type depending of the options:

```ts
export async function parse(
  input: string | BufReader
): Promise<string[][]>;
export async function parse(
  input: string | BufReader,
  opt: Omit<ParseOptions, "columns" | "skipFirstRow">,
): Promise<string[][]>;
export async function parse(
  input: string | BufReader,
  opt: Omit<ParseOptions, "columns"> & {
    columns: string[] | ColumnOptions[],
  },
): Promise<Record<string, unknown>[]>;
export async function parse(
  input: string | BufReader,
  opt: Omit<ParseOptions, "skipFirstRow"> & {
    skipFirstRow: true,
  },
): Promise<Record<string, unknown>[]>;
export async function parse(
  input: string | BufReader,
  opt: ParseOptions = {
    skipFirstRow: false,
  },
): Promise<string[][] | Record<string, unknown>[]> { ... }
```